pkg/controller/operconfig/migration.go

```diff
@@ -97,3 +128,382 @@ func convertEgressFirewallToEgressNetworkPolicy(ctx context.Context, client cnoc
+			if netnamespace.GetName() == namespace.Name {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func netNamespaceHasEgressIpConfig(nns uns.Unstructured) bool {
+	if nns.Object["egressIPs"] != nil {
+		return len(nns.Object["egressIPs"].([]interface{})) > 0
+	}
+	return false
+}
+
+func hostSubnetHasEgressIpConfigAutomatic(hsn uns.Unstructured) bool {
+	if hsn.Object["egressIPs"] != nil && hsn.Object["egressCIDRs"] != nil {
+		return len(hsn.Object["egressIPs"].([]interface{})) > 0 && len(hsn.Object["egressCIDRs"].([]interface{})) > 0
+	}
+	return false
+}
+
+func hostSubnetHasEgressIpConfigManual(hsn uns.Unstructured) bool {
+	if hsn.Object["egressIPs"] != nil && hsn.Object["egressCIDRs"] == nil {
+		return len(hsn.Object["egressIPs"].([]interface{})) > 0
+	}
+	return false
+}
+
+func labelNodeAndRemoveHostSubnetConfig(ctx context.Context, client cnoclient.Client, hsn *uns.Unstructured) error {
+	// label node as egressassignable
+	hostStr := fmt.Sprintf("%v", hsn.Object["host"])
+
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		nodeObj, err := client.Default().Kubernetes().CoreV1().Nodes().Get(ctx, hostStr, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		if nodeObj.Labels == nil {
+			nodeObj.Labels = make(map[string]string)
+		}
+		if _, ok := nodeObj.Labels[egressAssignable]; !ok {
+			nodeObj.Labels[egressAssignable] = ""
+		}
+
+		// copy egressCIDRs to node label if possible
+		// - if egressCIDRs contains values (automatic config), on rollback this annotation will provide the respective hostsubnet with its original egressCIDRs values
+		// - if egressCIDRs is empty (manual config), on rollback this label will not exist, and so we default to node's subnet for egressCIDR field.
+		egressCIDRs, ok := hsn.Object["egressCIDRs"]
+		if ok {
+			// automatic configuration
+			if nodeObj.Annotations == nil {
+				nodeObj.Annotations = make(map[string]string)
+			}
+			if _, ok := nodeObj.Annotations[egressCIDRAnnotationName]; !ok {
+				egressCIDRsText, err := json.Marshal(egressCIDRs)
+				if err != nil {
+					return err
+				}
+				nodeObj.Annotations[egressCIDRAnnotationName] = string(egressCIDRsText)
+			}
+		}
+		_, err = client.Default().Kubernetes().CoreV1().Nodes().Update(ctx, nodeObj, metav1.UpdateOptions{})
+		return err
+	}); err != nil {
+		return err
+	}
+
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		// now remove egressIP config from hostsubnet
+		hsn.Object["egressCIDRs"] = nil
+		hsn.Object["egressIPs"] = nil
+		_, err := client.Default().Dynamic().Resource(gvrHostSubnet).Update(ctx, hsn, metav1.UpdateOptions{})
+		if errors.IsConflict(err) {
+			// refresh hsn with Get() in the event of conflict
+			var getErr error
+			hsn, getErr = client.Default().Dynamic().Resource(gvrHostSubnet).Get(ctx, hsn.GetName(), metav1.GetOptions{})
+			if getErr != nil {
+				return getErr
+			}
+		}
+		return err
```

Comment:
Check the `RetryOnConflic` docs:
```
// RetryOnConflict is used to make an update to a resource when you have to worry about
// conflicts caused by other code making unrelated updates to the resource at the same
// time. fn should fetch the resource to be modified, make appropriate changes to it, try
// to update it, and return (unmodified) the error from the update function
```
You need to return the error from `Update` unmodified.
So the use pattern is how you used it earlier: first Get and then Update and return the error from Update.

Reply:
Please check other comments I had, I am not sure if they are being hidden from you when you marked the conversation as resolved.

Reply:
> You need to return the error from Update unmodified.
So the use pattern is how you used it earlier: first Get and then Update and return the error from Update.

I wanted to avoid potentially unecessary Get() calls, but I see the issue. I suppose an extra Get() call is harmless while the current code could lead to unforeseen issues. I'll make this change (here and also at another spot in the code)